TX2 build problem associated with gotcha

I'm building Caliper on ThunderX2 with openmpi 3.1.4 and gcc9.2. I run into this build error in the gotcha part, but not when building the recent versions of gotcha independently. GCC's recommended fix seems to resolve it.

```
[  3%] Building C object ext/gotcha/src/CMakeFiles/caliper-gotcha.dir/libc_wrappers.c.o
/home/gravelle/soft/src/Caliper/ext/gotcha/src/libc_wrappers.c: In function ‘gotcha_open’:
/home/gravelle/soft/src/Caliper/ext/gotcha/src/libc_wrappers.c:361:21: error: ‘SYS_open’ undeclared (first use in this function); did you mean ‘SYS_openat’?
  361 |    result = syscall(SYS_open, pathname, flags, mode);
      |                     ^~~~~~~~
      |                     SYS_openat
/home/gravelle/soft/src/Caliper/ext/gotcha/src/libc_wrappers.c:361:21: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [ext/gotcha/src/CMakeFiles/caliper-gotcha.dir/libc_wrappers.c.o] Error 1
make[1]: *** [ext/gotcha/src/CMakeFiles/caliper-gotcha.dir/all] Error 2
make: *** [all] Error 2
```

--- a/ext/gotcha/src/libc_wrappers.c
+++ b/ext/gotcha/src/libc_wrappers.c
@@ -358,7 +358,12 @@ int gotcha_open(const char *pathname, int flags, ...)
    }
    va_end(args);
 
+#if defined(SYS_open)
    result = syscall(SYS_open, pathname, flags, mode);
+#else
+   result = syscall(SYS_openat, AT_FDCWD, pathname, flags, mode);
+#endif
+
    if (result >= 0)
       return (int) result;
    

